src/App.tsx

```diff
@@ -26,70 +27,112 @@ type ReorderOptions = {
 
-  // Sort and reverse goods if needed
-  // eslint-disable-next-line no-console
-  console.log(sortType, isReversed);
+  visibleGoods.sort((goodA: string, goodB: string) => {
+    switch (sortType) {
+      case SortType.ALPHABET:
+        return goodA.localeCompare(goodB);
+
+      case SortType.LENGTH:
+        return goodA.length - goodB.length;
+
+      default:
+        return 0;
+    }
+  });
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
 
   return visibleGoods;
 }
 
-// DON'T save goods to the state
-// type State = {
-//   isReversed: boolean,
-//   sortType: SortType,
-// };
-
-export const App: React.FC = () => {
-  return (
-    <div className="section content">
-      <div className="buttons">
-        <button
-          type="button"
-          className="button is-info is-light"
-        >
-          Sort alphabetically
-        </button>
-
-        <button
-          type="button"
-          className="button is-success is-light"
-        >
-          Sort by length
-        </button>
-
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
-
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
-      </div>
+export class App extends PureComponent<{}, ReorderOptions> {
+  state: Readonly<ReorderOptions> = {
+    sortType: SortType.NONE,
+    isReversed: false,
+  };
+
+  handleSortClick = (sortType: SortType) => {
+    this.setState({ sortType });
+  };
+
+  handleReverseClick = () => {
+    this.setState(state => ({
+      isReversed: !state.isReversed,
+    }));
+  };
+
+  handleResetClick = () => {
+    this.setState({ sortType: SortType.NONE, isReversed: false });
+  };
+
+  render() {
+    const { sortType, isReversed } = this.state;
+    const isDefaultOrder = isReversed || sortType !== SortType.NONE;
+
+    const goods = getReorderedGoods(
+      goodsFromServer,
+      { sortType, isReversed },
+    );
+
+    return (
+      <div className="section content">
+        <div className="buttons">
+          <button
+            onClick={() => this.handleSortClick(SortType.ALPHABET)}
+            type="button"
+            className={cn('button is-info',
+              { 'is-light': sortType === SortType.ALPHABET })}
+          >
+            Sort alphabetically
+          </button>
+
+          <button
+            onClick={() => this.handleSortClick(SortType.LENGTH)}
+            type="button"
+            className={cn('button is-success',
+              { 'is-light': sortType === SortType.LENGTH })}
+          >
+            Sort by length
+          </button>
+
+          <button
+            onClick={this.handleReverseClick}
+            type="button"
+            className={cn('button is-warning',
+              { 'is-light': isReversed })}
+          >
+            Reverse
+          </button>
+
+          {isDefaultOrder && (
```

Comment:
It is confusing, because we should show reset when it isn't default order
```suggestion
          {shouldShowReset && (
```
or
```suggestion
          {isOrderChanged && (
```